third_party/hdf5.BUILD

```diff
@@ -0,0 +1,363 @@
+        "#define H5_HAVE_STRDUP 1",
+        "#define H5_HAVE_STRING_H 1",
+        "#define H5_HAVE_STRTOLL 1",
+        "#define H5_HAVE_STRTOULL 1",
+        "#define H5_HAVE_SYSTEM 1",
+        "#define H5_HAVE_SYS_STAT_H 1",
+        "#define H5_HAVE_SYS_TIMEB_H 1",
+        "#define H5_HAVE_SYS_TYPES_H 1",
+        "#define H5_HAVE_TIMEZONE 1",
+        "#define H5_HAVE_TMPFILE 1",
+        "#define H5_HAVE_ZLIB_H 1",
+        "#define H5_HAVE___INLINE 1",
+        "#define H5_INCLUDE_HL 1",
+        "#define H5_LDOUBLE_TO_LLONG_ACCURATE 1",
+        "#define H5_LLONG_TO_LDOUBLE_CORRECT 1",
+        '#define H5_LT_OBJDIR ".libs/"',
+        "#define H5_NO_ALIGNMENT_RESTRICTIONS 1",
+        '#define H5_PACKAGE "hdf5"',
+        '#define H5_PACKAGE_BUGREPORT "help@hdfgroup.org"',
+        '#define H5_PACKAGE_NAME "HDF5"',
+        '#define H5_PACKAGE_STRING "HDF5 1.10.6"',
+        '#define H5_PACKAGE_TARNAME "hdf5"',
+        '#define H5_PACKAGE_URL "http://www.hdfgroup.org"',
+        '#define H5_PACKAGE_VERSION "1.10.6"',
+        "#define H5_HAVE_PWD_H 1",
+        "#define H5_SIZEOF_BOOL 1",
+        "#define H5_SIZEOF_CHAR 1",
+        "#define H5_SIZEOF_DOUBLE 8",
+        "#define H5_SIZEOF_FLOAT 4",
+        "#define H5_SIZEOF_INT 4",
+        "#define H5_SIZEOF_INT16_T 2",
+        "#define H5_SIZEOF_INT32_T 4",
+        "#define H5_SIZEOF_INT64_T 8",
+        "#define H5_SIZEOF_INT8_T 1",
+        "#define H5_SIZEOF_INT_FAST64_T 8",
+        "#define H5_SIZEOF_INT_FAST8_T 1",
+        "#define H5_SIZEOF_INT_LEAST16_T 2",
+        "#define H5_SIZEOF_INT_LEAST32_T 4",
+        "#define H5_SIZEOF_INT_LEAST64_T 8",
+        "#define H5_SIZEOF_INT_LEAST8_T 1",
+        "#define H5_SIZEOF_LONG_LONG 8",
+        "#define H5_SIZEOF_SHORT 2",
+        "#define H5_SIZEOF_SIZE_T 8",
+        "#define H5_SIZEOF_TIME_T 8",
+        "#define H5_SIZEOF_UINT16_T 2",
+        "#define H5_SIZEOF_UINT32_T 4",
+        "#define H5_SIZEOF_UINT64_T 8",
+        "#define H5_SIZEOF_UINT8_T 1",
+        "#define H5_SIZEOF_UINT_FAST64_T 8",
+        "#define H5_SIZEOF_UINT_FAST8_T 1",
+        "#define H5_SIZEOF_UINT_LEAST16_T 2",
+        "#define H5_SIZEOF_UINT_LEAST32_T 4",
+        "#define H5_SIZEOF_UINT_LEAST64_T 8",
+        "#define H5_SIZEOF_UINT_LEAST8_T 1",
+        "#define H5_SIZEOF_UNSIGNED 4",
+        "#define H5_SIZEOF__QUAD 0",
+        "#define H5_STDC_HEADERS 1",
+        "#define H5_USE_110_API_DEFAULT 1",
+        '#define H5_VERSION "1.10.6"',
+        "#define H5_WANT_DATA_ACCURACY 1",
+        "#define H5_WANT_DCONV_EXCEPTION 1",
+        "",
+        "#endif",
+        "EOF",
+    ]),
+)
+
+genrule(
+    name = "H5lib_settings_c",
+    outs = ["config/H5lib_settings.c"],
+    cmd = "\n".join([
+        "cat <<'EOF' >$@",
+        "char H5libhdf5_settings[]=\"swiftnav\";",
+        "EOF",
+    ]),
+)
+
+genrule(
+    name = "H5Tinit_c",
+    outs = ["config/H5Tinit.c"],
+    cmd = "$(location :H5detect) $@",
```

Comment:
What is the content of the file this binary generates?

Reply:
https://pastebin.com/rjzdCntD

It has some machine/date specific things:
```
 * Created:		Jan  2, 2023
 *			Wiktor Więcław <wiktor@wiktor-Latitude-5521>
```

Reply:
I guess we could hardcode the source file if we find this to be a problem, right?

Reply:
Maybe we can do it in a separate PR but we will probably want to revisit this if causes problems with caching (the timestamps and user info will probably invalidate the cache).

Reply:
https://swift-nav.atlassian.net/browse/BUILD-499